pkg/chartutil/dependencies.go

```diff
@@ -106,6 +107,14 @@ func getAliasDependency(charts []*chart.Chart, dep *chart.Dependency) *chart.Cha
 		md := *c.Metadata
 		out.Metadata = &md
 
+		// empty dependencies and shallow copy all dependencies, otherwise parent info may be corrupted if
+		// there is more than one dependency aliasing this chart
+		out.SetDependencies()
```

Comment:
I seems from my local testing and the usecase I have (attached in zip) this is not enough while the following changes from this [other PR](https://github.com/helm/helm/pull/10174/files) is working:

```golang
		if md.Dependencies != nil {
			dependencies := make([]*chart.Dependency, len(md.Dependencies))
			for i := range md.Dependencies {
				d := *md.Dependencies[i]
				dependencies[i] = &d
			}
			md.Dependencies = dependencies
		}
```

[reproducer-common-dep-import-value.zip](https://github.com/helm/helm/files/8602843/reproducer-common-dep-import-value.zip)

Reply:
I am happy to look into this, when there is feedback from the maintainers what milestone is realistic for this PR. If there is a case I overlooked, I guess there should be another test case as well.

Reply:
@mederel since I had some spare time and these issues are relly quite annoying, I took a look at the issue you
mentioned.

The example from your zip boils down to the following graph:

```mermaid
graph TD;
    umbrella-->child1;
    umbrella-->child2;
    umbrella-->child2-bis;
    child2-bis-. alias .->child2;
    child1-->common;
    child2-->common;
    child2-bis-->common;
```

The example can be a bit simplified by removing `child1`, while still showing the issue

```mermaid
graph TD;
    umbrella-->child2;
    umbrella-->child2-bis;
    child2-bis-. alias .->child2;
    child2-->common;
    child2-bis-->common;
```

So we are essentially back to the test case from #9150, which is

```mermaid
graph TD;
    parent-->foo;
    parent-->bar;
    foo-. alias .->child;
    bar-. alias .->child;
    foo-->grandchild;
    bar-->grandchild;
```

The only notably difference is the use of `import-values` in the two child charts and this is where it gets interesting.

The issue reported in #9150 was related to information stored in
https://github.com/helm/helm/blob/ced54b12896ae933017782e2bbd722e2bc8c0914/pkg/chart/chart.go#L55
which is why I copied the `Chart` objects.

The issue you found is similar but in a different place:
https://github.com/helm/helm/blob/ced54b12896ae933017782e2bbd722e2bc8c0914/pkg/chart/dependency.go#L47
which is why you copied the `Dependency` objects in the metadata.

So im the end we need both. I updated this PR with a fix and a test case for the issue you mentioned.
I have to say that I don't like the solution. There seems to be a conceptual issue with aliases in Helm's dependency
management implementation from my point of view.

Some parts of the implementation seem to assume to reflect alias with the following graph in memory:

```mermaid
graph TD;
    parent-->child;
    parent-- aliased -->child;
```

while others assume

```mermaid
graph TD;
    parent-->child;
    parent-- aliased --> child-copy;
```

Depending on which path to follow information such as "imported values" or "parent" must be stored on the edges of the graph (first diagram) or can be stored on the nodes (second diagram). 

Anyway, there is no easy and nice way to fix this, but I would like to see a refactoring of the logic for Helm 4.